crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -195,4 +198,26 @@ def sanity_checks
       end
     end
   end
+
+  def upgrade
+    if request.post?
+      respond_to do |format|
+        format.html do
+          render file: "public/503_upgrade.html"
+        end
```

Comment:
But any user who starts an upgrade will not be surprised to be redirected to the upgrade page :-) Anyway, I don't mind much.